src/operator/nn/concat.cc

```diff
@@ -39,14 +39,16 @@ static bool ConcatShape(const nnvm::NodeAttrs& attrs,
   const ConcatParam& param_ = nnvm::get<ConcatParam>(attrs.parsed);
   CHECK_EQ(in_shape->size(), static_cast<size_t>(param_.num_args));
   TShape dshape;
-  index_t size = 0;
-  bool has_zero = false;
+  index_t size = 0, out_size = 0, has_zero = 0, zero_index = -1;
```

Comment:
More correctly, `size` and `out_size` should be `int64_t` since we have changed the type of the elements of `TShape` to `int64_t`.